app/src/main/java/com/firebase/chatapplication/di/providersModule.kt

```diff
@@ -3,13 +3,22 @@ package com.firebase.chatapplication.di
 import android.view.TextureView
 import com.firebase.chatapplication.MainActivity
 import com.firebase.chatapplication.managers.CameraXManager
+import com.firebase.chatapplication.managers.GeofenceManager
 import com.firebase.chatapplication.managers.IntentCameraManager
 import com.firebase.chatapplication.providers.SignInProvider
+import com.google.android.gms.location.LocationServices
 import org.koin.dsl.module.module
 
 val providersModule = module {
 
     single { (activity: MainActivity) -> SignInProvider(get(), get(), get(), activity) }
     single { IntentCameraManager() }
+    single { (activity: MainActivity) ->
+        GeofenceManager(
+            LocationServices.getGeofencingClient(
+                activity.applicationContext
+            ), activity.applicationContext
+        )
+    }
```

Comment:
This line can be replaced to:
`single { GeofenceManager(LocationServices.getGeofencingClient(get()), get()) }`

In Koin `aplicationContext` is part of module, so in can be injected directly by putting `get()`.
Activity context cannot be provided this way.

Reply:
Sounds cool!